spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -347,20 +347,23 @@ def sample_types_from(decoded_profile)
 
       let(:samples) { samples_from_pprof(encoded_pprof) }
 
+      def sample_allocation(obj, _unused = 0)
+        # Heap sampling currently requires this 2-step process to first pass data about the allocated object...
+        described_class::Testing._native_track_object(stack_recorder, obj, sample_rate)
+        Datadog::Profiling::Collectors::Stack::Testing
+          ._native_sample(Thread.current, stack_recorder, metric_values, labels, numeric_labels, 400, false)
+      end
+
       before do
         allocations = [a_string, an_array, 'a fearsome string', [-10..-1], a_hash, { 'z' => -1, 'y' => '-2', 'x' => false }]
         @num_allocations = 0
         allocations.each_with_index do |obj, i|
-          # Heap sampling currently requires this 2-step process to first pass data about the allocated object...
-          described_class::Testing._native_track_object(stack_recorder, obj, sample_rate)
-          # ...and then pass data about the allocation stacktrace (with 2 distinct stacktraces)
+          # Sample allocations with 2 distinct stacktraces
+          # (the 2nd parameter is there only to avoid a rubocop issue with identical branches)
```

Comment:
Minor: It's actually possible to shut rubocop up with an inline comment, e.g.

```ruby
        def self.pkg_config_missing?(command: $PKGCONFIG) # rubocop:disable Style/GlobalVars
```

... I guess it would be a bit cleaner here to do that than passing dummy args in?